### PR TITLE
[Test] Add pytest coverage for 30 example kernels

### PR DESCRIPTION
**Pull request by zwj-sep**
## 改动内容

为 `examples/` 和 `examples_experiment/` 下 30 个算子新增非侵入式 pytest 测试文件，不修改任何原文件。

### 覆盖范围

| 区域 | 类别 | 文件数 |
|------|------|--------|
| `examples/` | Activation | 10 |
| | Elementwise | 3 |
| | Reduce | 4 |
| | Softmax | 1 |
| | 简单 Vector | 12 |
| `examples_experiment/` | — | 3 |
| **合计** | | **30** |

### 两种策略

按原文件顶层执行风格选择：

- **importlib + argv 隔离**（16 文件）：有 `__main__` 守卫的文件，通过 `importlib.util` 加载模块并隔离 `sys.argv`，parametrize 原文件 config 为独立用例。
- **subprocess**（14 文件）：顶层 for 循环无 `__main__` 守卫的文件，用 `subprocess.run` 包装原文件执行，断言 `returncode == 0`。

### 验证结果

在 Ascend910 上验证：

```
100 tests collected
100 passed, 0 failed
```

ruff lint + format 全部通过。

### 备注

`hadamard_transform` 的 float16/bfloat16 用例已单独提修复 PR #1513（issue #1504），本 PR 只包含 float32 用例。

### CODE REVIEW

**Review by LLMZhangYC — DISMISSED** [review]
lgtm

**Comment by dlaerbinjwy** [comment]
一个小问题，建议可以斟酌修改，不修改合入的话除了单独进行覆盖率测试也不影响其他测试，有 18 个用例是跑在 `subprocess.run` 建议改成进程内，这个目前和我们的已有的test用例的风格并不统一，虽然并不影响CI流程，但是在bench_test里面单独执行覆盖率测试时就覆盖不到，算子登记进 manifest 后 bench 就改走 `pytest --cov` 了，覆盖率会掉到 0。同一个算子实测：

```
subprocess      0 行
importlib 直调  3634 行 / 144 文件
```

这个主要是担心覆盖率测试，本身用例直接提交跑CI应该是没问题的，我建议可以参考你 PR 里的，`test_example_cummin.py` 的 `_load_cummin_example()` 抄过来改个文件名即可。我这边给你改了一个`test_silu.py` 改完是这样：

```python
import importlib.util
from pathlib import Path
from types import ModuleType

import pytest
import torch


def _load_example() -> ModuleType:
    source = Path(__file__).with_name("silu.py")
    spec = importlib.util.spec_from_file_location("_silu_example_for_test", source)
    if spec is None or spec.loader is None:
        raise ImportError(f"Cannot load example module: {source}")

    module = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(module)
    return module


@pytest.mark.skipif(
    not hasattr(torch, "npu") or not torch.npu.is_available(),
    reason="requires an Ascend NPU",
)
def test_silu_run():
    _load_example()
```
这段我在 NPU 上跑过了，`1 passed`，ruff 过；

**Comment by zwj-sep** [comment]
感谢建议！已经将 18 个 subprocess 型测试全部改为 importlib 直调，参考 test_batch_gemm.py 的模式（importlib + argv 隔离）。更新已 push 到本 PR，覆盖率问题已解决。

**Review by ChaoyangJi — DISMISSED** [review]
lgtm

**Comment by dlaerbinjwy** [comment]
@zwj-sep 提醒一个事:examples/cummin/test_example_cummin.py 需要挪到 examples/cann-bench/cummin/ 下。

#1542 把 example_cummin.py 从 examples/cummin/ 移到了 examples/cann-bench/cummin/,现在 examples/cummin/ 目录里已经没有算子文件了。
直接把文件挪到 examples/cann-bench/cummin/test_example_cummin.py,内容不用改。这个路径也是登记表里的名字(我在 #1604 里把那两条重新指向了新位置)。

**Comment by zwj-sep** [comment]
感谢提醒！已经将 test_example_cummin.py 挪到 examples/cann-bench/cummin/ 下，内容不变。
